core/src/test/java/org/apache/iceberg/TestPartitionSpecParser.java

```diff
@@ -19,16 +19,20 @@
 
 package org.apache.iceberg;
 
+import org.apache.iceberg.types.Types;
 import org.junit.Assert;
 import org.junit.Test;
 
-public class TestPartitionSpecParser extends TableTestBase {
-  public TestPartitionSpecParser() {
-    super(1);
-  }
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+public class TestPartitionSpecParser {
+  private static final Schema SCHEMA = new Schema(
+      required(1, "id", Types.IntegerType.get()),
+      required(2, "data", Types.StringType.get())
+  );
 
   @Test
-  public void testToJsonForV1Table() {
+  public void testToJson() {
```

Comment:
Instead of changing existing test cases, let's add new test cases for the case you want to exercise.